### PR TITLE
DDF: Add support for all current NEO smart plugs with metering (TS011F)

### PR DESCRIPTION
**Pull request by iisworks**
I created his DDF to combine all NEO Plugs in one, it supports these rebranded Tuya TS011F plugs:  _TZ3000_gjnozsaz,  _TZ3000_w0qqde0g and  _TZ3000_zloso4jk. Increased minimum reporting time of 5s for frequently changing loads like a PC to prevent too frequent updates in deconz. It also includes FW reporting.

BTW: This file combines and can replace the existing ones (NAS-WR01B.json and NAS-WR01B_2.json. Not sure how to propose to replace the other existing DDF files?

### CODE REVIEW

**Comment by Smanar** [comment]
>BTW: This file combines and can replace the existing ones (NAS-WR01B.json and NAS-WR01B_2.json. NOt sure how to propose to replace the other existing DDF files?

If you delete a file in your fork, the "removal" will be in the PR too.

**Comment by iisworks** [comment]
Done, I know, but I just hope the other DDF creators don't mind their DDF was merged into a new one. Although I think having just one DDF where new manufacturer names can be added is much more convenient (also one of the DDFs was still in Silver mode). 

Can you please check if the DDF itself looks OK?

**Comment by Smanar** [comment]
Don't worry, all DDF are scanned at start and deconz restart from scratch.

On your DDF, you have different way to get the swversion

```
        {
          "name": "attr/swversion",
          "parse": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001", "eval": "Item.val = Attr.val"},
          "read": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001"}
        },
```

and

```
        {
          "name": "attr/swversion",
          "parse": {"at": "0x0001","cl": "0x0000","ep": 1, "eval": "Item.val = Attr.val", "fn": "zcl"}
        },
```

If I m right the more used code is

```
        {
          "name": "attr/swversion",
          "parse": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001", "script": "tuya_swversion.js"},
          "read": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001"}
        },

But to work the DDF need to be in the "tuya" folder for exemple
```

**Comment by iisworks** [comment]
Good catch. :)
I did that on purpose so it does not have to depend on the .js, which does not work well with only the DDF in the "user" devices folder. To me this is fine (FW will show up as 69 or 74 for example), or do you think it's better to use "tuya_swversion.js"?

That would be:
        {
          "name": "attr/swversion",
          "parse": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001", "script": "../tuya/tuya_swversion.js"},
          "read": {"fn": "zcl", "ep": 1, "cl": "0x0000", "at": "0x0001"}
        },

**Comment by Smanar** [comment]
Oups, right, I use bad path

So you are 100% right, can use the method you want, lot of DDF miss swversion and display "none"
And yes, it s possible too using the code in the "eval" (and the code is small)

But I think the person that have created the file tuya_swversion.js have made it for we use it as depency.

Will see the Manup decision ^^, but for me you can use the weapon you prefer ^^

**Comment by iisworks** [comment]
Can you send the exact parse line I should use to format it correctly without the .js?

**Comment by Smanar** [comment]
And BTW, good idea to increase the report for thoses plug https://forum.phoscon.de/t/nous-a1z-ddf-causes-high-cpu/2675

**Comment by iisworks** [comment]
Yeah, and I have seen this 1s update frequency in more DDF files, like those for Blitzwolf and Tuya. It would make the 3rd party app extremely busy processing device updates.

One thought: what happens when a manufacturer id is present in more than one DDF file? Will deconz just use the first it finds it in (and if so: how are DDFs processed)? I assume the /devices folder in user data has priority, but what if it is referenced in multiple DDF files under the built-in /devices: e.g. one in /devices/tuya and another in/devices/neo or /devices/blitzwolf?

**Comment by Smanar** [comment]
>One thought: what happens when a manufacturer id is present in more than one DDF file? Will deconz just use the first it finds it in (and if so: how are DDFs processed)?

Good question, can be the first or the last, but it need to not happen, I know some manufacture have clone, but I wait for all PR are merged to correct it.

**Comment by Smanar** [comment]
@iisworks I have a new clone https://github.com/dresden-elektronik/deconz-rest-plugin/issues/6459

If you can add it in the list too ?

**Comment by iisworks** [comment]
Added, although I think this is where it might become messy because it's not branded as a NEO plug as the device folder would suggest. There is a whole list of rebranded TS011F plugs that - also depending on firmware version - may or may not support reporting or jus tpolling. But if this DDF works reliably for most people I guess it's a good starting point. :)

**Comment by Smanar** [comment]
Yeah you are right, but I m tired of tuya clone, this one is a "NAS-WR01B" too, so I think it's a good idea to mix with your DDF.

But I think using only one DDF with 47 manufacture name just for them is worse.

**Comment by iisworks** [comment]
> Yeah you are right, but I m tired of tuya clone, this one is a "NAS-WR01B" too, so I think it's a good idea to mix it with your DDF, not the same brand, but there is different brand that sell same device, like "frient".
> 
> And I think using only one DDF with 47 manufacture name just for them is worse.
> 
> This one look standard, no firwmare special features/bugs.

What determines if a DDF will be included in the next release or not?

**Comment by Smanar** [comment]
No way to be sure, but from last timeline I had, the next PR merge session will be soon (someone is taking small holiday, so waiting for his return ^^), and as your PR is small I don't see a reason for it will be not in the package

File: devices/neo/NAS-WR01B_TS011F.json [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+    "_TZ3000_zloso4jk",
+    "_TZ3000_rdfh8cfs"
+  ],
+  "modelid": [
+    "TS011F",
+    "TS011F",
+    "TS011F",
+    "TS011F"
+  ],
+  "product": "NAS-WR01B",
```
Comment (SwoopX) [inline devices/neo/NAS-WR01B_TS011F.json:15]: Could you please add what kind of device this is? Just adding smart plug should suffice.

**Review by SwoopX — CHANGES_REQUESTED** [review]
Thanks for updating the DDF. You might want to add the relevant firmware code for the power sensor to have it consistent.

Additionally, I'd strongly advise to revisit the refresh intervals set. Generally, it is recommended to add like 60 seconds to the respectively set max reporting interval. E.g. for `state/voltage` with a max reporting interval, the refresh interval should be 360s.

**Comment by iisworks** [comment]
@SwoopX I updated the file, please check since I'm not too familiar with all of this. :)

**Comment by SwoopX** [comment]
@iisworks Hehe, no worries. That's fortunately just the minor details. If you could do the same for `state/on` and `state/consumption`, we should be good to go here

**Comment by iisworks** [comment]
@SwoopX I think I finally updated them all now. :)

**Review by SwoopX — APPROVED** [review]

**Comment by Smanar** [comment]
@iisworks don't worry the `"refresh.interval"` is generally just a security, you just need to use a value bigger than the one used in the report configuration.
The device need to use bind/report first, and if this one not working deconz will make a poll using the `"refresh.interval"`

If the device work fine, "refresh.interval" will be never used. If the report don't work 360s will be too much for a consumption report, this kind of sensor spam more, but don't worry for that if it have worked for you, will be fine.

**Pull request merged**